src/cdk/v2/destinations/the_trade_desk/transformConversion.js

```diff
@@ -0,0 +1,88 @@
+const { InstrumentationError } = require('@rudderstack/integrations-lib');
+const {
+  defaultRequestConfig,
+  simpleProcessRouterDest,
+  defaultPostRequestConfig,
+  removeUndefinedAndNullValues,
+} = require('../../../../v0/util');
+const { EventType } = require('../../../../constants');
+const { REAL_TIME_CONVERSION_ENDPOINT } = require('./config');
+const {
+  prepareFromConfig,
+  prepareCommonPayload,
+  getRevenue,
+  prepareItemsPayload,
+  getAdvertisingId,
+  prepareCustomProperties,
+  populateEventName,
+  getDataProcessingOptions,
+  getPrivacySetting,
+  enrichTrackPayload,
+} = require('./utils');
+
+const responseBuilder = (payload) => {
+  const response = defaultRequestConfig();
+  response.endpoint = REAL_TIME_CONVERSION_ENDPOINT;
+  response.method = defaultPostRequestConfig.requestMethod;
+  response.body.JSON = payload;
+  return response;
+};
+
+const validateInput = (message) => {
+  if (!message.type) {
+    throw new InstrumentationError('Event type is required');
+  }
+
+  const messageType = message.type.toLowerCase();
+  if (messageType !== EventType.TRACK) {
+    throw new InstrumentationError(`Event type "${messageType}" is not supported`);
+  }
+
+  if (!message.event) {
+    throw new InstrumentationError('Event is not present. Aborting.');
+  }
+};
+
+const prepareTrackPayload = (message, destination) => {
+  const configPayload = prepareFromConfig(destination);
+  const commonPayload = prepareCommonPayload(message);
+  const items = prepareItemsPayload(message);
+  const { id, type } = getAdvertisingId(message);
+  const customProperties = prepareCustomProperties(message, destination);
```

Comment:
what kind of custom properties can be expected from config ?
Are we not supporting TD1 to TD10 custom properties within event ?

Reply:
We are supporting it, TD1-TD10 should be mapped in  customProperties in destination config. In this utility we are just getting the mapped TD1-TD10 properties values.

Reply:
Is the requirement to have common custom properties across different event types ?

Reply:
no

Reply:
Creating a separate custom properties mapping for each event (separate + custom) will increase the connection setup time in webapp and if any event don't have the mapping in webapp then td1-td10 wouldn't be propogated to trade desk for that event.

Reply:
I'm not completely convinced on this functionality - can you sync up with @vthopu once ?

Reply:
We can release TD1-TD10 custom fields support as an enhancement post discussion to unblock from release

Reply:
Going with current functionality & supporting TD1-TD10 at individual event level will be an additional enhancement in next release